bigtable/tests/unit/test_table.py

```diff
@@ -589,6 +589,101 @@ def test_sample_row_keys(self):
         result = table.sample_row_keys()
         self.assertEqual(result[0], expected_result)
 
+    def test_truncate(self):
+        from google.cloud.bigtable_v2.gapic import bigtable_client
+        from google.cloud.bigtable_admin_v2.gapic import (
+            bigtable_table_admin_client)
+
+        data_api = bigtable_client.BigtableClient(mock.Mock())
+        table_api = bigtable_table_admin_client.BigtableTableAdminClient(
+            mock.Mock())
+        credentials = _make_credentials()
+        client = self._make_client(project='project-id',
+                                   credentials=credentials, admin=True)
+        client._table_data_client = data_api
+        client._table_admin_client = table_api
+        instance = client.instance(instance_id=self.INSTANCE_ID)
+        table = self._make_one(self.TABLE_ID, instance)
+
+        expected_result = None  # truncate() has no return value.
+
+        result = table.truncate()
+
+        self.assertEqual(result, expected_result)
```

Comment:
We need to make sure that the proto object that we expect is actually sent.  Is there a way to do that in unit tests?

Reply:
@sduskis Yes, there is.  The best would be to mock out the client's `_table_admin_client` (rather than mocking the real object's channel) and then just assert that its `drop_row_range` method had been called appropriately.  E.g.:

```python
    table_api = mock.create_autospec(
        bigtable_table_admin_client.BigtableTableAdminClient)
    client._table_admin_client = table_api
    ...
    table.truncate()
    ...
    table_api.drop_row_range.assert_called_once_with(
        name=self.TABLE_ID,
        delete_all_data_from_table=True,
        ...)
```